specs/deneb/beacon-chain.md

```diff
@@ -155,10 +157,24 @@ class ExecutionPayloadHeader(Container):
     withdrawals_root: Root
     data_gas_used: uint64  # [New in Deneb:EIP4844]
     excess_data_gas: uint64  # [New in Deneb:EIP4844]
+    parent_beacon_block_root: Root  # [New in Deneb:EIP4788]
 ```
 
 ## Helper functions
 
+### Beacon state accessors
+
+#### `get_previous_block_root`
+
+```python
+def get_parent_block_root(state: BeaconState) -> Root:
```

Comment:
It seems more natural to just read this from `state.latest_block_header.parent_root`. which would cover the genesis slot read without the conditional

and you get then probably get rid of this helper

Reply:
during block _construction_ `state.latest_block_header.parent_root` is the grandparent block root, which we don't want

I added this helper so we have a unified way to get the parent block root across all parts of the spec; if we want to drop it:

* during block construction, `hash_tree_root(state.latest_block_header)`
* during block validation, `state.latest_block_header.parent_root`

so we lose the compact nature of the helper but it does mean we don't have to worry about potentially underflowing subtraction

Reply:
I don't see this immediately being used elsewhere (e.g. the light client protocol) and I agree it seems simpler to drop this helper, so I'll go ahead and make this change

Reply:
this was a bit more involved than I was expecting and I'd check out the comment below but this change has been implemented now